src/Paprika/Data/SlottedArray.cs

```diff
@@ -24,15 +25,32 @@ namespace Paprika.Data;
 {
     private readonly ref Header _header;
     private readonly Span<byte> _data;
-    private readonly Span<Slot> _slots;
-    private readonly Span<byte> _raw;
 
     public SlottedArray(Span<byte> buffer)
     {
-        _raw = buffer;
-        _header = ref Unsafe.As<byte, Header>(ref _raw[0]);
+        _header = ref Unsafe.As<byte, Header>(ref MemoryMarshal.GetReference(buffer));
         _data = buffer.Slice(Header.Size);
-        _slots = MemoryMarshal.Cast<byte, Slot>(_data);
+    }
+
+    private readonly ref Slot this[int index]
```

Comment:
This is cool.

Reply:
Sequential is fine, just don't move fields 🙂